src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java

```diff
@@ -147,6 +153,10 @@ Map<String, Object> getResponseAsMap() throws IOException {
         return (Map<String, Object>) GSON.fromJson(responseAsString, Map.class);
       }
       if (statusCode == 429) {
+        String retryAfterTimeInSeconds =
+          Arrays.stream(response.getAllHeaders()).filter(t -> t.getName().equals(RETRY_AFTER)).
+            findFirst().get().getValue();
+        Thread.sleep(Integer.parseInt(retryAfterTimeInSeconds) * 1000);
```

Comment:
the retryer is already retrying based on the configured waitstrategy - https://github.com/data-integrations/zendesk/blob/17f796fa88f84077c0291f5baa917b13f497f738/src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java#L111. Adding a sleep here is just adding an additional delay